pkg/resources/spark/envoyfilter.go

```diff
@@ -0,0 +1,90 @@
+package spark
+
+import (
+	"fmt"
+
+	protobuftypes "github.com/gogo/protobuf/types"
+	"istio.io/api/networking/v1alpha3"
+	v1alpha32 "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	dcv1alpha1 "github.com/dominodatalab/distributed-compute-operator/api/v1alpha1"
+)
+
+const (
+	filterName = "envoy.filters.network.tcp_proxy"
+)
+
+// NewEnvoyFilter creates a new EnvoyFilter resource to set idle_timeout for Istio-enabled deployments
+func NewEnvoyFilter(sc *dcv1alpha1.SparkCluster) (v1alpha32.EnvoyFilter, error) {
```

Comment:
```suggestion
func NewEnvoyFilter(sc *dcv1alpha1.SparkCluster) *v1alpha32.EnvoyFilter {
```

You don't need to return an `error` since there's nothing in this func that can create one, right? Also, you may want to return a pointer, no?